presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoInstaller.java

```diff
@@ -73,15 +68,38 @@ public void runCommand()
     private static void installUsingRpm(String pathToRpm, boolean checkDependencies)
             throws PrestoManagerException
     {
-        int checkRpm = executeCommand("rpm", "-Kv", "--nosignature ", pathToRpm);
-        if (checkRpm != 0) {
-            throw new PrestoManagerException("Corrupted RPM", checkRpm);
-        }
+        checkRpmPackage(pathToRpm);
         String nodeps = checkDependencies ? "" : "--nodeps";
-        int installRpm = executeCommand(150, "sudo", "rpm", "-iv", nodeps, pathToRpm);
+        int installRpm = executeCommand(SUBPROCESS_TIMEOUT, "sudo", "rpm", "-iv", nodeps, pathToRpm);
         if (installRpm != 0) {
             throw new PrestoManagerException("Failed to install presto", installRpm);
         }
         LOGGER.debug("Successfully installed presto");
     }
+
+    static File fetchFileFromUrl(URL urlToFetchPackage)
```

Comment:
This is very, very similar to a method in `AgentFileUtils`. Unfortunately, it is different enough that that method can't be used here, but we should make note of that for a future update, where we can make `AgentFileUtils` a more robust class and use it more widely.